src/app/App.tsx

```diff
@@ -0,0 +1,219 @@
+   */
+  useEffect(() => {
+    const queryFilters   = getFilterStateFromQuery();
+    const queryClassroom = getClassroomStateFromQuery();
+    const queryBuilding  = getBuildingStateFromQuery();
+
+    if ( false === _.isEqual(filters, queryFilters) ) {
+      // console.log('setting filters');
+      setFilters(queryFilters);
+    }
+
+    if ( false === _.isEqual(classroom, queryClassroom) ) {
+      // console.log('setting classroom');
+      setClassroom(queryClassroom);
+    }
+
+    if ( false === _.isEqual(building, queryBuilding) ) {
+      // console.log('setting building');
+      setBuilding(queryBuilding);
+    }
+  }, [searchParams]);
+
+  const getFilterStateFromQuery = () => {
+    try {
+      const filterString = searchParams.get('filters') || '{}';
+      const filters      = JSON.parse(decodeURIComponent(filterString));
+
+      return filters;
+    } catch(e) {
+      console.error('Error parsing filters:', e);
+      return {};
+    }
+  }
+
+  const getClassroomStateFromQuery = () => {
+    try {
+      let classroomSlug = searchParams.get('classroom') || null;
+
+      if ( classroomSlug ) {
+        classroomSlug = decodeURIComponent(classroomSlug);
+      }
+
+      return classroomSlug ?? '';
+    } catch(e) {
+      console.error('Error parsing classroom slug:', e);
+      return '';
+    }
+  }
+
+  const getBuildingStateFromQuery = () => {
+    try {
+      let buildingSlug = searchParams.get('building') || null;
+
+      if ( buildingSlug ) {
+        buildingSlug = decodeURIComponent(buildingSlug);
+      } 
+
+      return buildingSlug ?? '';
+    } catch(e) {
+      console.error('Error parsing building slug:', e);
+      return '';
+    }
+  }
+
+  const [filters, setFilters] = useState(getFilterStateFromQuery());
+  const [classroom, setClassroom] = useState(getClassroomStateFromQuery());
+  const [building, setBuilding] = useState(getBuildingStateFromQuery());
+  const [loading, setLoading] = useState(false);
+
+  const handleShowClassroom = (newClassroom) => {
+    const filterString = encodeURIComponent(newClassroom);
+    setSearchParams({...searchParams, classroom: filterString });
+  }
+
+  const handleShowBuilding = (newBuilding) => {
+    const filterString = encodeURIComponent(newBuilding);
+    setSearchParams({...searchParams, building: filterString });
+  }
+  
+  const handleClearBuilding = () => {
+    setSearchParams({...searchParams, building: '' });
+  }
+
+  const handleClearClassroom = () => {
+    setSearchParams({...searchParams, classroom: '' });
+  }
+
+  const setFilterQueryState = (newFilters) => {
+    const filterString = encodeURIComponent(JSON.stringify(newFilters));
+    setSearchParams({...searchParams, filters: filterString });
+  }
+
+  const handleSubmitFilters = (newFilters) => {
+    setFilterQueryState(newFilters);
+  }
+
+  const handleClearFilters = () => {
+    setFilterQueryState({});
+  }
+
+  console.log({
```

Comment:
Are we leaving these in place as we do the acceptance testing? Or can the console.log be removed?

Reply:
I would like to leave these logs for now, as we are still in a testing phase. Thanks!